File: sdkwork-backend-sdk-go/api/knowledge_base.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type KnowledgeBaseApi struct {
    client *sdkhttp.Client
}

func NewKnowledgeBaseApi(client *sdkhttp.Client) *KnowledgeBaseApi {
    return &KnowledgeBaseApi{client: client}
}

// Update an existing knowledge base
func (a *KnowledgeBaseApi) Update(body sdktypes.PlusKnowledgeBaseForm) (sdktypes.PlusApiResultPlusKnowledgeBaseVO, error) {
    raw, err := a.client.Put(BackendApiPath("/knowledge_base"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusKnowledgeBaseVO](raw)
}

// Create a new knowledge base
func (a *KnowledgeBaseApi) Create(body sdktypes.PlusKnowledgeBaseForm) (sdktypes.PlusApiResultPlusKnowledgeBaseVO, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusKnowledgeBaseVO](raw)
}

// Get knowledge bases by page
func (a *KnowledgeBaseApi) ListByPage(body *sdktypes.QueryListForm, query map[string]interface{}) (sdktypes.PlusApiResultPagePlusKnowledgeBaseVO, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/list"), body, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPagePlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPagePlusKnowledgeBaseVO](raw)
}

// Get all knowledge bases
func (a *KnowledgeBaseApi) ListAllEntities(body *sdktypes.QueryListForm) (sdktypes.PlusApiResultListPlusKnowledgeBaseVO, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/list/all"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListPlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListPlusKnowledgeBaseVO](raw)
}

// Get a knowledge base detail by ID
func (a *KnowledgeBaseApi) GetDetail(query map[string]interface{}) (sdktypes.PlusApiResultPlusKnowledgeBaseVO, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/get_detail"), nil, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultPlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusKnowledgeBaseVO](raw)
}

// List files
func (a *KnowledgeBaseApi) ListFiles(query map[string]interface{}) (sdktypes.PlusApiResultFileListVO, error) {
    raw, err := a.client.Get(BackendApiPath("/knowledge_base/files"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultFileListVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultFileListVO](raw)
}

// Upload file
func (a *KnowledgeBaseApi) UploadFile(body *sdktypes.UploadFilePostRequest, query map[string]interface{}) (sdktypes.PlusApiResultFileItemVO, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/files"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.PlusApiResultFileItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultFileItemVO](raw)
}

func (a *KnowledgeBaseApi) Stop(query map[string]interface{}, headers map[string]string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/chat/stop"), nil, query, headers, "")
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// Create a chat completion with Knowledge base
func (a *KnowledgeBaseApi) CreateCompletions(body sdktypes.ChatCompletionCreateForm, query map[string]interface{}, headers map[string]string) (sdktypes.ChatCompletionChunk, error) {
    raw, err := a.client.Post(BackendApiPath("/knowledge_base/chat/completions"), body, query, headers, "")
    if err != nil {
        var zero sdktypes.ChatCompletionChunk
        return zero, err
    }
    return decodeResult[sdktypes.ChatCompletionChunk](raw)
}

// Get a knowledge base by ID
func (a *KnowledgeBaseApi) GetById(id string) (sdktypes.PlusApiResultPlusKnowledgeBaseVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/knowledge_base/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPlusKnowledgeBaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPlusKnowledgeBaseVO](raw)
}

// Delete a knowledge base
func (a *KnowledgeBaseApi) Delete(id string) (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/knowledge_base/%s", id)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// Get file
func (a *KnowledgeBaseApi) GetFile(fileId string) (sdktypes.PlusApiResultFileItemVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/knowledge_base/files/%s", fileId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultFileItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultFileItemVO](raw)
}

// Delete file
func (a *KnowledgeBaseApi) DeleteFile(fileId string) (sdktypes.PlusApiResultFileItemVO, error) {
    raw, err := a.client.Delete(BackendApiPath(fmt.Sprintf("/knowledge_base/files/%s", fileId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultFileItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultFileItemVO](raw)
}

// Get file content
func (a *KnowledgeBaseApi) GetFileContent(fileId string) (string, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/knowledge_base/files/%s/content", fileId)), nil, nil)
    if err != nil {
        var zero string
        return zero, err
    }
    return decodeResult[string](raw)
}

File: sdkwork-backend-sdk-typescript/src/api/knowledge-base.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { ChatCompletionChunk, ChatCompletionCreateForm, PlusApiResultBoolean, PlusApiResultFileItemVO, PlusApiResultFileListVO, PlusApiResultListPlusKnowledgeBaseVO, PlusApiResultPagePlusKnowledgeBaseVO, PlusApiResultPlusKnowledgeBaseVO, PlusKnowledgeBaseForm, QueryListForm } from '../types';


export class KnowledgeBaseApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Update an existing knowledge base */
  async update(body: PlusKnowledgeBaseForm): Promise<PlusApiResultPlusKnowledgeBaseVO> {
    return this.client.put<PlusApiResultPlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base`), body);
  }

/** Create a new knowledge base */
  async create(body: PlusKnowledgeBaseForm): Promise<PlusApiResultPlusKnowledgeBaseVO> {
    return this.client.post<PlusApiResultPlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base`), body);
  }

/** Get knowledge bases by page */
  async listByPage(body?: QueryListForm, params?: QueryParams): Promise<PlusApiResultPagePlusKnowledgeBaseVO> {
    return this.client.post<PlusApiResultPagePlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base/list`), body, params);
  }

/** Get all knowledge bases */
  async listAllEntities(body?: QueryListForm): Promise<PlusApiResultListPlusKnowledgeBaseVO> {
    return this.client.post<PlusApiResultListPlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base/list/all`), body);
  }

/** Get a knowledge base detail by ID */
  async getDetail(params?: QueryParams): Promise<PlusApiResultPlusKnowledgeBaseVO> {
    return this.client.post<PlusApiResultPlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base/get_detail`), undefined, params);
  }

/** List files */
  async listFiles(params?: QueryParams): Promise<PlusApiResultFileListVO> {
    return this.client.get<PlusApiResultFileListVO>(backendApiPath(`/knowledge_base/files`), params);
  }

/** Upload file */
  async uploadFile(body?: FormData, params?: QueryParams): Promise<PlusApiResultFileItemVO> {
    return this.client.post<PlusApiResultFileItemVO>(backendApiPath(`/knowledge_base/files`), body, params);
  }

async stop(params?: QueryParams, headers?: Record<string, string>): Promise<PlusApiResultBoolean> {
    return this.client.post<PlusApiResultBoolean>(backendApiPath(`/knowledge_base/chat/stop`), undefined, params, headers);
  }

/** Create a chat completion with Knowledge base */
  async createCompletions(body: ChatCompletionCreateForm, params?: QueryParams, headers?: Record<string, string>): Promise<ChatCompletionChunk> {
    return this.client.post<ChatCompletionChunk>(backendApiPath(`/knowledge_base/chat/completions`), body, params, headers);
  }

/** Get a knowledge base by ID */
  async getById(id: string | number): Promise<PlusApiResultPlusKnowledgeBaseVO> {
    return this.client.get<PlusApiResultPlusKnowledgeBaseVO>(backendApiPath(`/knowledge_base/${id}`));
  }

/** Delete a knowledge base */
  async delete(id: string | number): Promise<PlusApiResultBoolean> {
    return this.client.delete<PlusApiResultBoolean>(backendApiPath(`/knowledge_base/${id}`));
  }

/** Get file */
  async getFile(fileId: string | number): Promise<PlusApiResultFileItemVO> {
    return this.client.get<PlusApiResultFileItemVO>(backendApiPath(`/knowledge_base/files/${fileId}`));
  }

/** Delete file */
  async deleteFile(fileId: string | number): Promise<PlusApiResultFileItemVO> {
    return this.client.delete<PlusApiResultFileItemVO>(backendApiPath(`/knowledge_base/files/${fileId}`));
  }

/** Get file content */
  async getFileContent(fileId: string | number): Promise<string> {
    return this.client.get<string>(backendApiPath(`/knowledge_base/files/${fileId}/content`));
  }
}

export function createKnowledgeBaseApi(client: HttpClient): KnowledgeBaseApi {
  return new KnowledgeBaseApi(client);
}
